rtrlib/rtr_mgr.c

```diff
@@ -58,21 +59,24 @@ static int rtr_mgr_start_sockets(struct rtr_mgr_group *group)
 	return RTR_SUCCESS;
 }
 
-int rtr_mgr_find_group(const struct rtr_mgr_config *config,
-		       const struct rtr_socket *sock,
-		       unsigned int *ind)
+static struct rtr_mgr_group *rtr_mgr_find_group(struct rtr_mgr_config *config,
+						const struct rtr_socket *sock)
 {
-	for (unsigned int i = 0; i < config->len; i++) {
-		for (unsigned int j = 0;
-		     j < config->groups[i].sockets_len; j++) {
-			if (config->groups[i].sockets[j] == sock) {
-				*ind = i;
-				return RTR_SUCCESS;
-			}
+	tommy_node *node;
+	struct rtr_mgr_group_node *group_node;
+
+	node  = tommy_list_head(&config->groups);
+
+	while (node) {
+		group_node = node->data;
+		for (unsigned int j = 0; j < group_node->group->sockets_len;
```

Comment:
why isn't `sockets_len` of type `size_t` anyway? Or alternatively just `unsigned` should suffice, too.